include/mata/ord_vector.hh

```diff
@@ -470,6 +470,14 @@ public:   // Public methods
 		return vec_.empty();
 	}
 
+	inline bool back() const
+	{
+		// Assertions
+		assert(vectorIsSorted());
+
+		return vec_.back();
+	}
```

Comment:
We need to get the element (with its transitions), not just `bool`.

Reply:
Furthermore, calling `back()` on an empty vector causes UB.

Either we have to always first check whether the given state has any transitions at all (our back would cause UB, too, following a behaviour of `std::vector<T>::back()`), or we must implement some logic handling this case.

We could return a pointer, which we would fill with `nullptr` in the case the vector is empty. We could also create a method `Mata::Nfa::Nfa::get_epsilon_transitions()`, which would return a `const_iterator` to `TransitionList` which would either be an iterator to the epsilon transitions, or `TransitionList::end()` in case there are no epsilon transitions for the given state.

I strongly recommend using the first option:
- follow the behaviour of `std::vector<T>::back()` with returning `T&` from `OrdVector::back()`,
- create method `get_epsilon_transitions()` which checks for emptiness of `TransitionList` and either returns `TransitionList::end()` if there are no epsilon transitions, or return a `const_iterator` to the epsilon transitions.

Reply:
I agree. In OrdVector we should follow behaviour of underlying data structure. And add the `get_epsilon_transitions()` method to Nfa.

Reply:
The logic seems OK. I have refactorized the function further in #69. Closing this conversation as resolved.